perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java

```diff
@@ -135,7 +220,14 @@ public boolean consentHubExists(PerunSession sess, ConsentHub consentHub) {
 
 	@Override
 	public void deleteConsentHub(PerunSession perunSession, ConsentHub consentHub) throws ConsentHubAlreadyRemovedException {
-		//TODO: remove all user consents first
+
+		try {
+			for (Consent consent : getConsentsForConsentHub(perunSession, consentHub.getId())) {
```

Comment:
I think deleting all the consent hub's consents should be moved to the BL layer because the `deleteConsent` method in the BL layer should also audit the deletion message. If it remained here, we would lose the audit information.